pkg/task/restore.go

```diff
@@ -131,33 +147,55 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		int64(len(ranges)+len(files)),
 		!cfg.LogProgress)
 
-	err = restore.SplitRanges(ctx, client, ranges, rewriteRules, updateCh)
+	clusterCfg, err := restorePreWork(ctx, client, mgr)
 	if err != nil {
-		log.Error("split regions failed", zap.Error(err))
 		return err
 	}
 
+	// Do not reset timestamp if we are doing incremental restore, because
+	// we are not allowed to decrease timestamp.
 	if !client.IsIncremental() {
 		if err = client.ResetTS(cfg.PD); err != nil {
 			log.Error("reset pd TS failed", zap.Error(err))
 			return err
 		}
 	}
 
-	removedSchedulers, err := restorePreWork(ctx, client, mgr)
-	if err != nil {
-		return err
+	// Restore sst files in batch.
+	batchSize := int(math.Min(maxRestoreBatchSizeLimit /* 256 */, float64(cfg.Concurrency)))
```

Comment:
```suggestion
	batchSize := cfg.Concurrency
	if batchSize > maxRestoreBatchSizeLimit {
		batchSize = maxRestoreBatchSizeLimit // 256
	}
```

or use `mathutil.Min(cfg.Concurrency, maxRestoreBatchSizeLimit)` if you must need this to be single-line.